test/externalTests/plugins/testCommon.js

```diff
@@ -77,136 +78,124 @@ function inject (bot) {
         if (expectedBlock === null) {
           if (block.name === 'air') continue
           // dig it
-          return digAndResume(position)
+          await dig(position)
         } else {
           if (expectedBlock.type === block.type) continue
           // fix it
           if (block.type !== 0) {
             // dig it
-            return digAndResume(position)
+            await dig(position)
           }
           console.log('going to place layer ', y, 'with item ', superflatLayers[y].item.type, position)
           // place it
-          return placeAndResume(position, superflatLayers[y].item)
+          await place(position, superflatLayers[y].item)
         }
       }
     }
-    // all good
-    cb()
 
-    function digAndResume (position) {
-      bot.dig(bot.blockAt(position), resume)
+    function dig (position) {
+      return bot.dig(bot.blockAt(position))
     }
 
-    function placeAndResume (position, item) {
+    async function place (position, item) {
       // console.log('place and resume with', item)
-      setInventorySlot(36, new Item(item.type, 1, 0), () => {
-        placeBlock(36, position, resume)
-      })
-    }
-
-    function resume () {
-      resetBlocksToSuperflat(cb)
+      await setInventorySlot(36, new Item(item.type, 1, 0))
+      await placeBlock(36, position)
     }
   }
 
-  function placeBlock (slot, position, cb) {
+  async function placeBlock (slot, position) {
     bot.setQuickBarSlot(slot - 36)
     // always place the block on the top of the block below it, i guess.
     const referenceBlock = bot.blockAt(position.plus(new Vec3(0, -1, 0)))
-    bot.placeBlock(referenceBlock, new Vec3(0, 1, 0), cb)
+    return bot.placeBlock(referenceBlock, new Vec3(0, 1, 0))
   }
 
   // always leaves you in creative mode
-  function resetState (cb) {
-    callbackChain([
-      becomeCreative,
-      clearInventory,
-      (cb) => {
-        // console.log('start flying')
-        bot.creative.startFlying()
-        teleport(new Vec3(0, 4, 0), cb)
-      },
-      cb => bot.waitForChunksToLoad(cb),
-      resetBlocksToSuperflat,
-      (cb) => { setTimeout(cb, 1000) },
-      clearInventory
-    ], cb)
+  async function resetState () {
+    await becomeCreative()
+    await clearInventory()
+    bot.creative.startFlying()
+    await teleport(new Vec3(0, 4, 0))
+    await bot.waitForChunksToLoad()
+    await resetBlocksToSuperflat()
+    await sleep(1000)
+    await clearInventory()
```

Comment:
You can use `Promise.all` to join stuff that can be executed in parallel

Reply:
I'm aware, but I simply ported the code from the callback version. It was not clear which actions can be parallel. As there are two clearInventory calls, there is clearly a defined order

Reply:
```js
    await becomeCreative()
    bot.creative.startFlying()
    await Promise.all([
      clearInventory(),
      teleport(new Vec3(0, 4, 0))
    ])
    await bot.waitForChunksToLoad()
    await resetBlocksToSuperflat()
    await sleep(1000)
    await clearInventory()
```

or
```js
    await becomeCreative()
    bot.creative.startFlying()
    await Promise.all([
      clearInventory(),
      teleport(new Vec3(0, 4, 0)),
      bot.waitForChunksToLoad()
    ])
    await resetBlocksToSuperflat()
    await sleep(1000)
    await clearInventory()
```